substrate/frame/nomination-pools/src/lib.rs

```diff
@@ -3133,7 +3133,13 @@ impl<T: Config> Pallet<T> {
 			// the sum of the pending rewards must be less than the leftover balance. Since the
 			// reward math rounds down, we might accumulate some dust here.
 			let pending_rewards_lt_leftover_bal = RewardPool::<T>::current_balance(id) >=
-				pools_members_pending_rewards.get(&id).copied().unwrap_or_default();
+				pools_members_pending_rewards
+					.get(&id)
+					.copied()
+					.unwrap_or_default()
+					// allow for some tiny tolerance here to account for existential deposit
+					// increases which may only be handled lazily.
+					.saturating_sub(T::Currency::minimum_balance());
```

Comment:
This will get rid of the errors in short term but we should also put in a fix so it does not happen in future before closing the issue. I would suggest instead to temporarily comment out the ensure check on L3145 with a link to  https://github.com/paritytech/polkadot-sdk/issues/158 if you don't want the try state checks to fail for now. It will still spit out the warnings so thats good. I will try to put out a solution for the fix in a separate PR by this week.

Reply:
ED could increase or decrease. I think having a check that has a 0.1% tolerance or ED (which ever is larger in that case) would be good. Maybe even 2x ED or so?  
Eventually it will still fail when ED does a 10x or so, but then we already know where to look.

Reply:
Maybe I misunderstood. When the ED changes, the imbalance created in a pool can exceed the new ED value?

Reply:
For now, yes, actually not ED but the difference of (current_ed - last_ed) which is 3_000_000_000.

But I think Oliver's point is, in future ED might increase again causing the deficit to add up. 

I still think though this is only a short term tolerance and in long term we should fix the issue and strive to never have deficit at all in the pool. That's how the pool was designed as well and the ED increase issue is a bug we found because of the try state check. We would not want to ignore another potential issue with the pool by keeping these tolerance checks long term.

Reply:
> But I think Oliver's point is, in future ED might increase again causing the deficit to add up. Assuming we fix the issue before the next ED increase, we should be fine with a lower tolerance.

In the future if the ED goes up, `T::Currency::minimum_balance` should also go up so wouldn't that be OK?

I hear you though regarding missing potential issues with pools from this check. What we could do is make this log a warning if there's an imbalance below the tolerance for now, and in your PR fixing the root remove the tolerance allowance entirely. Lmk if that sounds like a good compromise to you.

Reply:
Had a chat with Kian and agreed on keeping the warning but not panicking since this error is not caused by new code. So the changes could be just removing the [`ensure` check](https://github.com/paritytech/polkadot-sdk/pull/1236/files#diff-fd1d9327e2dde740889545cb51e8e116c70b311d4318fd031cb95735ffdbb936L3145).

Reply:
@Ank4n I have replaced the ensure check with a warning.